ImportTool/RequestGenerator.cs

```diff
@@ -0,0 +1,120 @@
+
+                (int count, int succeedCount) = await AddInputPartsFromBlobs(cloudBlobClient, prefix, maxFileNumber, parameters);
+
+                Console.WriteLine($"Total files: {count} traversed.");
+                Console.WriteLine($"Succeed files: {succeedCount} put as input.");
+            }
+
+            return new FhirJsonSerializer().SerializeToString(parameters);
+        }
+
+        private static async Task<Tuple<int, int>> AddInputPartsFromBlobs(CloudBlobClient cloudBlobClient, string prefix, int maxFileNumber, Parameters parameters)
+        {
+            int count = 0;
+            int succeed_count = 0;
+
+            BlobContinuationToken continuationToken = null;
+            do
+            {
+                var segments = await cloudBlobClient.ListBlobsSegmentedAsync(
+                    prefix: prefix,
+                    useFlatBlobListing: true,
+                    blobListingDetails: BlobListingDetails.Metadata,
+                    maxResults: Math.Min(maxFileNumber, 5000),
+                    currentToken: continuationToken,
+                    options: null,
+                    operationContext: null);
+
+                foreach (var segment in segments.Results.Cast<CloudBlockBlob>())
+                {
+                    try
+                    {
+                        string resourceType = await GetResourceType(segment);
+                        List<Tuple<string, Base>> inputPart = new List<Tuple<string, Base>>();
+                        inputPart.Add(Tuple.Create("type", (Base)new FhirString(resourceType)));
+                        inputPart.Add(Tuple.Create("url", (Base)new FhirUri(segment.Uri)));
+                        string etag = segment.Properties.ETag.Replace("\"", string.Empty, StringComparison.OrdinalIgnoreCase);
+                        inputPart.Add(Tuple.Create("etag", (Base)new FhirString(etag)));
+                        parameters.Add("input", inputPart);
+                        succeed_count++;
+                    }
+                    catch (Exception oex)
+                    {
+                        Console.WriteLine($"Error reslove file {segment.Uri} due to exception {oex.Message}");
+                    }
+
+                    count++;
+                }
+            }
+            while (continuationToken != null && count < maxFileNumber);
+
+            return new Tuple<int, int>(count, succeed_count);
+        }
+
+        // Get the first resouce in file to determine type
+        private static async Task<string> GetResourceType(CloudBlockBlob cloudBlockBlob)
+        {
+            using (var stream = await cloudBlockBlob.OpenReadAsync())
+            {
+                char[] buffer = ArrayPool<char>.Shared.Rent(128);
+                try
+                {
+                    using (StreamReader reader = new StreamReader(stream))
+                    {
+                        reader.Read(buffer, 0, 128);
```

Comment:
we can read first line

Reply:
First line may be very large, like imageStudy.